clippy_lints/src/collapsible_if.rs

```diff
@@ -54,24 +76,19 @@ declare_clippy_lint! {
     /// Should be written:
     ///
     /// ```rust.ignore
-    /// if x && y {
-    ///     …
-    /// }
-    ///
-    /// // or
-    ///
     /// if x {
     ///     …
     /// } else if y {
     ///     …
     /// }
     /// ```
-    pub COLLAPSIBLE_IF,
+    pub COLLAPSIBLE_ELSE_IF,
     style,
```

Comment:
Having the split out lint in the `style` category will result in people having to allow this lint again in their code, if they don't like it. 

I guess the argument that this is fine, is that this case of the lint is easier to fix than the other case?

Reply:
Yeah, hopefully it won't be too annoying. :/
I also think that the  `if {} else { if {} } ` warning is actually more useful than the `if { if {} }` warning.

Reply:
Looking at your Rust PR, where you linked to this, it seems Joshua also agrees. So I'm ok with this.